File: C_Programming/Practise_Programs/PP_100.c
/*
==============================================================================
 File Name   : PP_100.c

 Program     : Display a Character Array as a String

 Description : Initializes a character array with individual characters
               and displays the array as a string using the standard
               output function.

 Author      : Shubham Gadhe

 Time Complexity  : O(n)
 Space Complexity : O(1)
==============================================================================
*/

#include<stdio.h>

int main()
{
    char str[] = {'J','a','y',' ','G', 'a', 'n' , 'e' , 's' , 'h'};

    // Display the character array as a string.
    printf("%s\n", str);        

    return 0;
}

/*
==============================================================================
 Sample Output
------------------------------------------------------------------------------
Undefined Output

Note:
- The character array is not null-terminated ('\0').
- Since printf("%s", str) expects a null-terminated string, the behavior
  of this program is undefined.
==============================================================================
*/

File: Java_Programming/Practise_Programs/PP_100.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_100
//
//  Description       : This program demonstrates the error caused
//                      by placing a semicolon after the for loop.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 12/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class PP_100
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int Arr[] = {10,20,30,40,50};

        // Bad Programming Practise
        for(int iCnt = 0; iCnt < Arr.length; iCnt++); 
        {
            System.out.println(Arr[iCnt]);
        }

        System.out.println(iCnt);   // Error
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      Compilation Error
//
//////////////////////////////////////////////////////////////////